### PR TITLE
Upgrade python version to 3.9

### LINKED ISSUE #1412: Upgrade to python 3.8
## Abstract

Upgrade to python 3.8

## Task

- [ ] Make sure that we can get python 3.8 in all OSes we build for.
- [ ] Since the blocker that was pyinstaller support for python 3.8 is [fixed](https://github.com/pyinstaller/pyinstaller/issues/4311) bump pyinstaller and try to build binaries with python 3.8

### PR DESCRIPTION
**Pull request by yabirgb**
Closes #1412

### CODE REVIEW

File: setup.py [hunk 1]
```diff
@@ -28,7 +28,7 @@
         # TODO: Investigate if it's needed. rotkehlchen.spec is where files seem to be copied
         'rotkehlchen': ['data/*.json', 'data/*.meta', 'data/*.db'],
     },
-    python_requires='>=3.6',
+    python_requires='>=3.9.6',
```
Comment (LefterisJP) [inline setup.py:31]: Would we really require >=3.9.6 though?

Would nobody be able to run it with 3.9.1? OR 3.8? Gotta be careful here.

**Comment by yabirgb** [comment]
I didn't test the dockerfile but seems to have created the image without errors. I just updated the python base image

**Comment by kelsos** [comment]
macOS binary seems fine

File: rotkehlchen/tests/pylint/test_lognokwargs.py [hunk 2]
```diff
@@ -19,7 +19,7 @@ def test_simple_logger_with_kwargs(pylint_test_linter):
 
         checker.visit_call(node)
         messages = checker.linter.release_messages()
-        assert len(messages) == 1
+        assert len(messages) == 2
```
Comment (LefterisJP) [inline rotkehlchen/tests/pylint/test_lognokwargs.py:22]: This is probably a bug in either pylint or in our module. That's not theright solution ;)

**Review by LefterisJP — APPROVED** [review]
Hmmm one thing about the pylint custom module we got ... but the rest seem fine

**Pull request merged**